src/controllers/action-queue/util/makeExecutionContext.ts

```diff
@@ -0,0 +1,13 @@
+import { EdgeAccount } from 'edge-core-js'
+
+import { checkActionEffect } from '../runtime/checkActionEffect'
+import { evaluateAction } from '../runtime/evaluateAction'
+import { ExecutionContext } from '../types'
+
+export const makeExecutionContext = (properties: { account: EdgeAccount; clientId: string }): ExecutionContext => {
+  return {
+    ...properties,
+    evaluateAction,
+    checkActionEffect
```

Comment:
Can you bind the two methods to this instance, so you don't need to pass the context redundantly?

```ts
const out = {
 account,
 clientId,
 evaluateAction(program, state) {
   return evaluateAction(out, program, state)
 },
 checkActionEffect(effect) {
   return checkActionEffect(out, effect)
 }
}
return out
```